packages/errors/rejector.js

```diff
@@ -0,0 +1,72 @@
+import { Fail } from './index.js';
+
+const makeReject = (prePre, postPre) => {
+  /**
+   * @param {TemplateStringsArray | string[]} template
+   * @param {...any} args
+   * @returns {never}
+   */
+  const reject = ([...template], ...args) => {
+    for (;;) {
+      switch (typeof postPre) {
+        case 'string': {
+          break;
+        }
+        case 'number':
+        case 'bigint': {
+          // based on throw-labeled.js
+          postPre = `[${postPre}]`;
+          break;
+        }
+        default: {
+          postPre = `${postPre}`;
+          break;
+        }
+      }
+      template[0] = `${postPre}${template[0]}`;
+      if (prePre === undefined) {
+        break;
+      }
+      [prePre, postPre] = prePre;
+    }
+    throw Fail(template, ...args);
+  };
+  /**
+   * @param {string|number} prefix
+   * @returns {typeof reject}
+   */
+  reject.pre = prefix => makeReject([prePre, postPre], prefix);
```

Comment:
This method name is visible. Either it should not be (best) or it should be a good name. Since no one outside this package will normal need to be aware of it, there's no penalty for a long name.

Reply:
done